### PR TITLE
Entity View

### PR DESCRIPTION
**Pull request by ViktorBasanets**
A model was created and based on it - controller, service and dao, as well as the corresponding unit tests for the controller

### CODE REVIEW

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -343,36 +343,36 @@ spring.resources.chain:
...
-  jpa:
-    hibernate:
-      ddl-auto: "validate"
-    database-platform: "${SPRING_JPA_DATABASE_PLATFORM:org.hibernate.dialect.HSQLDialect}"
-  datasource:
-    driverClassName: "${SPRING_DRIVER_CLASS_NAME:org.hsqldb.jdbc.JDBCDriver}"
-    url: "${SPRING_DATASOURCE_URL:jdbc:hsqldb:file:${SQL_DATA_FOLDER:/tmp}/thingsboardDb;sql.enforce_size=false;hsqldb.log_size=5}"
-    username: "${SPRING_DATASOURCE_USERNAME:sa}"
-    password: "${SPRING_DATASOURCE_PASSWORD:}"
+# spring:
```
Comment (ashvayka) [inline application/src/main/resources/thingsboard.yml]: No need to change this file.

File: application/src/test/java/org/thingsboard/server/controller/BaseDeviceControllerTest.java [hunk 2]
```diff
@@ -41,7 +41,9 @@
 import org.junit.Test;
 
 import com.fasterxml.jackson.core.type.TypeReference;
-
+/**
+ * Created by Victor Basanets on 8/27/2017.
```
Comment (ashvayka) [inline application/src/test/java/org/thingsboard/server/controller/BaseDeviceControllerTest.java]: This is not valid statement :-) .

File: common/data/src/main/java/org/thingsboard/server/common/data/EntityView.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    public EntityView() {
+        super();
+    }
+
+    public EntityView(EntityViewId id) {
+        super(id);
+    }
+
+    public EntityView(EntityId entityId,
```
Comment (ashvayka) [inline common/data/src/main/java/org/thingsboard/server/common/data/EntityView.java]: Let's use @AllArgumentConstructor?

File: common/data/src/main/java/org/thingsboard/server/common/data/EntityView.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        this.keys = keys;
+        this.tsStart = tsStart;
+        this.tsEnd = tsEnd;
+    }
+
+    public EntityView(EntityView entityView) {
+        super(entityView);
+    }
+
+    public EntityId getEntityId() {
```
Comment (ashvayka) [inline common/data/src/main/java/org/thingsboard/server/common/data/EntityView.java]: @Data annotation may help to reduce a lot of boilerplate code. We may refactor all other similar classes in the future.

File: dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    public EntityView unassignEntityViewFromCustomer(EntityViewId entityViewId) {
+        EntityView entityView = findEntityViewById(entityViewId);
+        entityView.setCustomerId(null);
+        return saveEntityView(entityView);
+    }
+
+    @Override
+    public void deleteEntityView(EntityViewId entityViewId) {
+        log.trace("Executing deleteEntityView [{}]", entityViewId);
+        Cache cache = cacheManager.getCache(ENTITY_VIEW_CACHE);
```
Comment (ashvayka) [inline dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java]: Whare is the caching of this entities?

File: application/src/main/resources/thingsboard.yml (hunk 1 above)
Comment (ViktorBasanets) [reply application/src/main/resources/thingsboard.yml]: Done!

File: application/src/test/java/org/thingsboard/server/controller/BaseDeviceControllerTest.java (hunk 2 above)
Comment (ViktorBasanets) [reply application/src/test/java/org/thingsboard/server/controller/BaseDeviceControllerTest.java]: Done!

Comment (ViktorBasanets) [reply application/src/test/java/org/thingsboard/server/controller/BaseDeviceControllerTest.java]: This happened by accident! :)

File: dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java [hunk 6]
```diff
@@ -38,9 +38,7 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
```
Comment (ViktorBasanets) [inline dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java]: was deleted as useless

File: dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java [hunk 7]
```diff
@@ -38,9 +38,7 @@
...
 
-import static org.thingsboard.server.common.data.CacheConstants.ENTITY_VIEW_CACHE;
```
Comment (ViktorBasanets) [inline dao/src/main/java/org/thingsboard/server/dao/entityview/EntityViewServiceImpl.java]: was deleted as useless

**Pull request merged**